File: snippets/code/FishHave/Bubble sort/bubble_sort.java
public static <Type>[] bubbleSort(<Type>[] arr) {
    int len = arr.length;
    
    for (int i = 0; i < len - 1; i++) {
        for (int j = 0; j < len - 1 - i; j++) {
            if (arr[j] > arr[j + 1]) {
                <Type> temp = arr[j+1];
                arr[j+1] = arr[j];
                arr[j] = temp;
            }
        }
    }

    return arr;
}

File: snippets/code/FishHave/Bubble sort/bubble_sort.go
func sort(arr <Type>[]) <Type>[] {
	len := len(arr);

    for i := 0; i < len - 1; i++ {
        for j := 0; j < len - 1 - i; j++ {
            if arr[j] > arr[j + 1] {
                temp := arr[j+1];
                arr[j+1] = arr[j];
                arr[j] = temp;
        }
    }

    return arr;
} 